src/compiler/data.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+require('../globals');
+
+module.exports = data;
+
+var vlfield = require('../field');
+
+function data(encoding) {
+  var def = [data.raw(encoding)];
+
+  // TODO(kanitw): Aug 8, 2015 - if aggregate add TABLE
+  def.push(data.aggregated(encoding));
```

Comment:
With Vega 2, we still need to split the two .. But we  have to do less work in terms of sorting